pkg/platform/kube/platform.go

```diff
@@ -466,6 +468,64 @@ func (p Platform) EnrichFunctionConfig(ctx context.Context, functionConfig *func
 		functionConfig.Spec.PriorityClassName = p.Config.Kube.DefaultFunctionPriorityClassName
 	}
 
+	if p.Config.Kube.PreemptibleNodes != nil {
+
+		// we do such stuff to allow exposing features before they are exposed on UI
+		if preemptionMode, exists := functionConfig.Meta.Annotations["custom.nuclio.io/preemptible-mode"]; exists {
+			p.Logger.DebugWithCtx(ctx,
+				"Enriching function preemption mode from function annotations",
+				"preemptionMode", preemptionMode)
+			functionConfig.Spec.PreemptionMode = functionconfig.RunOnPreemptibleNodeMode(preemptionMode)
+		}
+
+		if functionConfig.Spec.PreemptionMode == functionconfig.RunOnPreemptibleNodesAllow {
+			p.Logger.DebugWithCtx(ctx,
+				"Allowing function to run on preemptible nodes",
+				"functionName", functionConfig.Meta.Name)
+
+			// merge node selectors - precedence to pre-configured preemptible nodes node selectors
+			if p.Config.Kube.PreemptibleNodes.NodeSelectors != nil {
+				functionConfig.Spec.NodeSelector = labels.Merge(functionConfig.Spec.NodeSelector,
+					p.Config.Kube.PreemptibleNodes.NodeSelectors)
+			}
+
+			// union tolerations
+			if len(p.Config.Kube.PreemptibleNodes.Tolerations) > 0 {
+
+				// initialize list
+				if functionConfig.Spec.Tolerations == nil {
+					functionConfig.Spec.Tolerations = []v1.Toleration{}
+				}
+
+				// only add
+				var tolerationsToAdd []v1.Toleration
+
+				// only add non-matching toleratinos as we dont want to have duplicates
+				for _, functionToleration := range functionConfig.Spec.Tolerations {
+					for _, preemptibleNodeTolerations := range p.Config.Kube.PreemptibleNodes.Tolerations {
+						if !functionToleration.MatchToleration(&preemptibleNodeTolerations) {
+							continue
+						}
+					}
+				}
```

Comment:
You don't add the `NodeTolerations` to `tolerationsToAdd`, only skipping duplicates.

Reply:
will update comment 👍🏼